### DISCUSSION TITLE
Underscore in test method is throwing the error

### DISCUSSION THREAD

**Discussion by akarthik-Ferguson** [pmd/pmd#3406] [Q&A]
Hi, 

PMD is throwing the error on the test method which contains "_" . 
`@IsTest
public class Builder_Test {

    @IsTest
    static void leadBuilder_build() {}
}`

How do I customize the rule and reference in my ruleset.xml?
`<rule ref="category/apex/codestyle.xml">
	<rule ref="category/apex/codestyle.xml/MethodNamingConventions">
    <properties>
        <property name="testPattern" value="[a-z_][a-zA-Z_0-9]*"/>
    </properties>
	</rule>`
The above one is not working.

**Comment by oowekyala** [comment]
So I still haven't gotten around to check this, but possibly the rule doesn't associate `@IsTest` with the `testPattern` pattern? To me there's nothing wrong with your regex, and it looks like all the tests of the rule use `@isTest` with no capital `I`...
